Dockerfile

```
@@ -1,5 +1,19 @@
-FROM openjdk:11-jdk-slim
+# Stage 1 : build
+FROM quay.io/quarkus/ubi-quarkus-native-s2i:22.1-java11 AS build
 
-COPY build/libs/kafka-processor-cosmos-block-*-standalone.jar /opt/kafka-processor-cosmos-block.jar
+COPY --chown=quarkus:quarkus . /usr/src/app
 
-ENTRYPOINT ["java","-jar","/opt/kafka-processor-cosmos-block.jar"]
+WORKDIR /usr/src/app
+
+RUN ./gradlew build -Dquarkus.package.type=native --no-daemon
```

We can bypass the test task here, it is already covered by the Test workflow.

```suggestion
RUN ./gradlew build -x check -Dquarkus.package.type=native --no-daemon
```